src/server/middleware/csrf-token.js

```diff
@@ -0,0 +1,67 @@
+import uuid from 'node-uuid';
+
+export const generateToken = (req, res, next) => {
+  // Don't generate new tokens for API calls
+  if (req.path.match(/^\/api/)) {
+    return next();
+  }
+
+  // TODO: Remove debug
+  const csrfToken = uuid.v4();
+  if (typeof req.session.csrfTokens === 'undefined') {
+    req.session.csrfTokens = [];
+  }
+
+  req.session.csrfTokens.push(csrfToken);
+
+  // Only 5 valid CSRF tokens at a time. Safe to increase number.
+  while (req.session.csrfTokens.length > 5) {
+    req.session.csrfTokens.shift();
+  }
```

Comment:
@cjwatson: I think this is a good idea, but would be better introduced by a feature further down the road.